kdc/kerberos5.c

```diff
@@ -1860,6 +1861,36 @@ generate_pac(astgs_request_t r, const Key *skey, const Key *tkey)
 	return ret;
     }
 
+    will_canonicalize = r->req.req_body.kdc_options.canonicalize ||
+	r->client->entry.flags.force_canonicalize;
+
+    /*
+     * For non-synthetic principals which match the realm of the issued
```

Comment:
What about synthetic clients though?  Why isn't their name canonical?

Reply:
Per out-of-band conversation, because they're ephemeral. But, really, because I'm concerned this path might break TktBridgeAP. So yes, this check can probably be removed.